mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -62,6 +92,14 @@ abstract class Classifier[
   /** @group setParam */
   def setRawPredictionCol(value: String): E = set(rawPredictionCol, value).asInstanceOf[E]
 
+  override def fit(dataset: Dataset[_]): M = {
+    // This handles a few items such as schema validation.
+    // Developers only need to implement train().
+    transformSchema(dataset.schema, logging = true)
+    copyValues(train(dataset).setParent(this))
+      .setPredictionMetadata(generatePredictionMetadata(dataset.schema))
```

Comment:
Here, you could pass in numClasses, which is now available from the model.  generatePredictionMetadata could take an optional numClasses argument.